### ISSUE TITLE
sql: use like operator with JSON and Inverted Indexes for matching

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#50345]
We've had a recent customer request for executing queries with a like operator on JSON properties for matching. For example:

```SQL
SELECT * FROM jsontable 
WHERE jcolum->'name' LIKE 'Andy%';
```

Today, this query does not use the inverted index. We could speed it up if it could take advantage of this query.

After discussing with @jordanlewis I believe this to be an optimizer task that relates to @sumeerbhola's recent work.

Jira issue: CRDB-4131

**Comment by awoods187** [comment]
cc @kai-niemi 

@RaduBerinde would this be something we could consider if we have extra time in 20.2?

**Comment by RaduBerinde** [comment]
Yes, possibly. CC @mjibson who has been working on some things related to inverted indexes.

**Comment by sumeerbhola** [comment]
The `invertedFilterer` processor can execute these if the relevant optimizer changes are made. @rytaft did the optimizer changes for the `invertedFilterer` for geospatial and left a TODO here https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/opt/xform/custom_funcs.go#L976-L977

**Comment by RaduBerinde** [comment]
I don't think we even need the inverted filterer in this case. It's a simple non-overlapping union, since the same object can't show up in multiple index entries for `jcolumn->'name'`.

**Comment by jordanlewis** [comment]
Just to note, we need to exclude the `-> int` form because items at particular array indexes aren't indexed by our scheme. Looks like the optimizer already ignores these cases so maybe nothing to do here.

**Comment by RaduBerinde** [comment]
CC @angelazxu

**Comment by rytaft** [comment]
still relevant

**Comment by kibertoad** [comment]
Are there still plans to implement this?

**Comment by rytaft** [comment]
It's not on our near-term roadmap, but we hope to get to it at some point. I added it to this meta-issue: https://github.com/cockroachdb/cockroach/issues/59331.

@kibertoad if you share some details about your use case that might help us to prioritize this.
